src/Core/Checkout/Customer/SalesChannel/SwitchDefaultAddressRoute.php

```diff
@@ -97,13 +106,19 @@ public function swap(string $addressId, string $type, SalesChannelContext $conte
                     'defaultBillingAddressId' => $addressId,
                 ];
 
+                $event = new CustomerSetDefaultBillingAddressEvent($context, $customer);
```

Comment:
This event should pass what the new id is 🤔 . Otherwise the user of the subscriber needs to refetch the Customer

Reply:
Here $customer is the customer Entity object so it could be fetched directly

Reply:
the passed $customer object does not contain the new IDs. 

the data objects are not in sync with the DB. so you would need to fetch the customer anew in your event subscriber.